encrypted-media-respec.html

```diff
@@ -606,7 +606,83 @@
           <section>
             <h5>Get Supported Configuration</h5>
             <p>Given a <a def-id="keysystems"></a> implementation <var>implementation</var>, <a>MediaKeySystemConfiguration</a> <var>candidate configuration</var>, and <var>origin</var>, this algorithm returns a supported configuration or <code>null</code> as appropriate.</p>
-            <p class="note">Unrecognized dictionary members in <var>candidate configuration</var> are ignored per [[WebIDL]] and will never reach this algorithm. Thus, they cannot be considered as part of the configuration.</p>
+            <p class="note">Unrecognized dictionary members in <var>candidate configuration</var> are ignored per [[WebIDL]] and will never reach this algorithm. Thus, they cannot be considered as part of the configuration.
+            </p>
+            <ol>
+              <li>
+                <dl class="switch">
+                  <dt>
+                    If the <a def-id="option-distinctiveIdentifier"></a> member of
```

Comment:
Do you mean to say we should completely generalize the consent mechanism ? Presently we talk explicitly about "consent for use of distinctive identifiers with _configuration_". Since the distinctive identifier state is part of _configuration_ we could generalize and simply talk about "consent to use _configuration_" and then in the leaf "get consent status" algorithm we would implement the one normative requirement (that distinctive identifier consent is required unless all the recommendations of the security section are followed).

I would also add a note about User Agents having flexibility to determine the set of _configuration_ values that a given consent applies to, since this depends on the language that was used to obtain the consent and UA decisions on how granular to be about asking for consent for different things.

We then still have to deal with the "optional" cases. We could punt this to the application (remove the `optional` states and force the application to list the alternatives it would accept. But the `optional` states do avoid having to deal with a combinatoric explosion at the application: instead we end up having to try every combination in our algorithm to find one that works.